### PR TITLE
Bug: Included check for payload.user and userEmail for Zendesk

### PR DESCRIPTION
**Pull request by abandonned**
## Description of the change
Included check for payload.user and throwing error if userEmail is not found.

> Notion -> https://www.notion.so/rudderstacks/Zendesk-Aellocred-6b12bb67dafd484e8b4188feaa6685f5

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: __tests__/data/zendesk_input.json [hunk 1]
```diff
@@ -1,4 +1,65 @@
...
+        "locale": "en-GB",
+        "os": {
+          "name": "",
+          "version": ""
+        },
+        "screen": {
+          "density": 2
+        },
+        "traits": {
+        },
```
Comment (saikumarrs) [inline __tests__/data/zendesk_input.json:47]: For track events, there is also an edge case where a missing `name` field in `traits` can cause an exception (while trying to create a new user). So, can we modify this test case input exactly to cover the missing `email` case?

Comment (abandonned) [reply __tests__/data/zendesk_input.json:47]: this test case is for the first error, where if none of the traits in identify mapping is present user is not created inside payload.
We can add test case to show the 2nd error as well.

Comment (saikumarrs) [reply __tests__/data/zendesk_input.json:47]: For the first error, just the absence of `email` is enough. I believe this data is good enough for the second error but might need more work to cover that.

Comment (abandonned) [reply __tests__/data/zendesk_input.json]: They both are of different eventType (track and identify) therefore included a new test case to demonstrate email misssing in traits error.

Comment (saikumarrs) [reply __tests__/data/zendesk_input.json]: I was referring to the exception thrown in `createUser` method.

Comment (abandonned) [reply __tests__/data/zendesk_input.json]: Sorry. `createUser` method is being in two different places. Inside `processTrack` and in `processGroup`. 

- For `processTrack` if we don't have `userEmail` we are throwing error so it won't reach `createUser` without `userEmail` . And we don't want to make call without a valid `userEmail` .
- For `processGroup` there is already a check present for `traits.name` and `traits.email`

Comment (saikumarrs) [reply __tests__/data/zendesk_input.json]: Agreed. In order to cover the first error, the input should just not contain `email` field and for the second one (in `createUser`), at least the `name` field should be missing in the `traits` object.
We should have test cases to explicitly cover each of them.

**Review by saikumarrs — APPROVED** [review]
